hail/python/hailtop/batch_client/aioclient.py

```diff
@@ -535,7 +535,7 @@ def _create_job(self,
                     in_update_parent_ids.append(job._job_id)
             else:
                 assert isinstance(job, SubmittedJob)
-                if self._batch is None or job._batch != self._batch:
+                if self._batch is None or (job._batch != self._batch and job._batch.id != self._batch.id):
```

Comment:
I think the `job._batch != self._batch` part of this expression is redundant now and can be deleted.